lang/c/src/datafile.c

```diff
@@ -568,6 +568,93 @@ int avro_file_reader_fp(FILE *fp, const char *path, int should_close,
 	return 0;
 }
 
+int
+avro_file_reader_json_schema(const char *file_path, const char **json_schema)
+{
+	FILE *file = fopen(file_path, "rb");
+	if (!file) {
+		avro_set_error("Error opening file: %s",
+			       strerror(errno));
+		return errno;
+	}
+
+	avro_reader_t reader = avro_reader_file(file);
```

Comment:
call to `avro_reader_free` makes sure closing the file.

Reply:
are we supposed to call this in failure cases/other return statements?